src/cmake-tools.ts

```diff
@@ -271,34 +287,52 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
       rollbar.invokeAsync('Changing CMake kit', async () => {
         if (kit) {
           log.debug('Injecting new Kit into CMake driver');
-          const drv = await this._cmakeDriver;
-          await drv.setKit(kit);
+          const drv = await this.getCMakeDriverInstance();
+          if(drv) {
+            await drv.setKit(kit);
+          }
         }
         this._statusBar.setActiveKitName(kit ? kit.name : '');
       });
     });
     this._ctestController.onTestingEnabledChanged(enabled => { this._statusBar.ctestEnabled = enabled; });
     this._ctestController.onResultsChanged(res => { this._statusBar.testResults = res; });
 
-    // Listen for terminal closing, we should create a new one on next launch
-    vscode.window.onDidCloseTerminal((term) => {
-      if (term == this._launchTerminal) {
-        this._launchTerminal = null;
-      }
-    });
-
-    // Finally, start the CMake driver
-    const drv = await (this._cmakeDriver = this._startNewCMakeDriver());
-    // Reload any test results. This will also update visibility on the status
-    // bar
-    await this._ctestController.reloadTests(drv);
     this._statusBar.setStatusMessage('Ready');
-    this._statusBar.targetName = this.defaultBuildTarget || await this.allTargetName;
 
     // Additional, non-extension: Start up nagging.
     this._nagManager.start();
   }
 
+  /**
+   * Returns, if possible a cmake driver instance. To creation the driver instance,
+   * there are preconditions that should be fulfilled, such as an active kit is selected.
+   * These preconditions are checked before it driver instance creation. When creating a
+   * driver instance, this function waits until the driver is ready before returning.
+   * This ensures that user commands can always be executed, because error criterials like
+   * exceptions would assign a null driver and it is possible to create a new driver instance later again.
+   */
+  async getCMakeDriverInstance() : Promise<CMakeDriver | null> {
+    if (!this._kitManager.hasActiveKit) {
+      return null;
+    }
+
+    if ((await this._cmakeDriver) == null) {
+      try {
+        this._cmakeDriver = this._startNewCMakeDriver();
+        let cmakeInstance = await this._cmakeDriver;
+        // Reload any test results. This will also update visibility on the status
+        // bar
+        await this._ctestController.reloadTests(cmakeInstance!);
```

Comment:
I think you should put `await this._cmakeDriver!` directly into the function call to avoid creating an unneeded variable.

Reply:
I think this is more readable.